tests/test_data_deletion/test_final_deletion.py

```diff
@@ -161,14 +173,24 @@ def test_delete(self, mocked_archive_project, mocked_archive_run, mocked_deletab
         mocked_archive_run.assert_any_call('a_run')
         mocked_archive_run.assert_any_call('another_run')
 
-    @patch('egcg_core.rest_communication.get_documents', return_value=run_elements1)
+    @patch('egcg_core.rest_communication.get_documents', return_value=full_run)
     @patch('egcg_core.rest_communication.get_document', return_value=sample1)
     def test_try_archive_run(self, mocked_get_doc, mocked_get_docs):
         assert os.path.exists(os.path.join(self.deleter.fastq_dir, 'a_run'))
         assert not os.path.exists(os.path.join(self.deleter.run_archive_dir, 'a_run'))
         self.deleter._try_archive_run('a_run')
         assert not os.path.exists(os.path.join(self.deleter.fastq_dir, 'a_run'))
         assert os.path.exists(os.path.join(self.deleter.run_archive_dir, 'a_run'))
+        mocked_get_docs.assert_called_once_with('run_elements', all_pages=True, where={'run_id': 'a_run'})
+        mocked_get_doc.assert_any_call('samples', where={'sample_id': 'a_sample1'})
+        mocked_get_doc.assert_any_call('samples', where={'sample_id': 'a_sample2'})
+        mocked_get_doc.assert_any_call('samples', where={'sample_id': 'a_sample3'})
+        mocked_get_doc.assert_any_call('samples', where={'sample_id': 'a_sample4'})
+        mocked_get_doc.assert_any_call('samples', where={'sample_id': 'a_sample5'})
+        mocked_get_doc.assert_any_call('samples', where={'sample_id': 'a_sample6'})
+        mocked_get_doc.assert_any_call('samples', where={'sample_id': 'a_sample7'})
+        mocked_get_doc.assert_any_call('samples', where={'sample_id': 'a_sample8'})
+        # Undetermined is ignored
```

Comment:
We're asserting that `get_document` has been called for the 8 samples, but we're not asserting that it hasn't been called for Undetermined. Something like `assert mocked_get_doc.call_count == 8` might help?